pkg/observability/observability.go

```diff
@@ -0,0 +1,79 @@
+package observability
+
+import (
+	"go.opencensus.io/exporter/prometheus"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	by            = "By"
+	ms            = "ms"
+	dimensionless = "1"
+)
+
+// Known exporters
+const (
+	AzureMonitor = "azure_monitor"
+	Datadog      = "datadog"
+	Jaeger       = "jaeger"
+	Prometheus   = "prometheus"
+	Stackdriver  = "stackdriver"
+	Zipkin       = "zipkin"
+)
+
+// PromExporter is the prometheus exporter containing HTTP handler for "/metrics"
+var PromExporter *prometheus.Exporter
+
+// Tags
+var (
+	KeyHostname, _               = tag.NewKey("hostname")
+	KeyJWTValidationErrorType, _ = tag.NewKey("error")
+)
+
+// Metrics
+var (
+	MRequestsByHostname         = stats.Int64("opencensus.io/http/proxy/request_total_by_host", "Number of proxied requests by target hostname", dimensionless)
+	MJWTManagerValidationErrors = stats.Int64("opencensus.io/plugin/jwt_manager/validation_error_total", "Number of validation errors by error type", dimensionless)
+	MOAuth2MissingHeader        = stats.Int64("opencensus.io/plugin/oauth2/missing_header_total", "Number of failed oauth2 authentication due to missing header", dimensionless)
+	MOAuth2MalformedHeader      = stats.Int64("opencensus.io/plugin/oauth2/malformed_header_total", "Number of failed oauth2 authentication due to malformed bearer header", dimensionless)
+	MOAuth2Authorized           = stats.Int64("opencensus.io/plugin/oauth2/authorized_request_total", "Number of successful and authorized oauth2 authentication", dimensionless)
+	MOAuth2Unauthorized         = stats.Int64("opencensus.io/plugin/oauth2/unauthorized_request_total", "Number of successful but unauthorized oauth2 authentication", dimensionless)
+)
+
+// AllViews aggregates the metrics
+var AllViews = []*view.View{
+	{
+		Name:        "opencensus.io/http/proxy/request_total_by_host",
```

Comment:
Would it be helpful to have proxy request latency by the host (or, maybe even by the API Name)?
The total amount of requests by host show us only usage information, not the performance of the proxy.

Reply:
API name and THHP method would be even better

Reply:
As a example @s4nji it will be as follows on dashboards:

*Incoming 99th Request Latency by Path Breakdown*
- `/api/menus/*` 500ms
- `/api/recipes/*` 100ms

When http client metrics enabled we would have the following:

*Outgoing 99th Request Latency by Path Breakdown*
- `http://api-v2.live-k8s.hellofresh.io/menus` 500ms
- `http://api-v2.live-k8s.hellofresh.io/recipes` 100ms

Reply:
@spetrashov can you refer to @s4nji the OC guide-lines you wrote so he can expose the same metrics as other applications. Having that we can leverage the http://grafana.live-k8s.hellofresh.io/d/hellofresh-services/services-traffic dashboards to see services traffic

Reply:
These metrics are now added via a9482e1, be38cb0, and 87ab607!